engine/realtimelogger.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <concurrentqueue.h>
+
+#include <fmt/core.h>
+#include <thread>
+
+namespace pong
+{
+
+static constexpr int MAX_LOG_MESSAGE_SIZE = 512;
+
+struct LogData
+{
+    int mLevel { 0 };
+    int mSequenceNum { 0 };
+    char mMessage[MAX_LOG_MESSAGE_SIZE] {};
+};
+
+class RealTimeLogger
+{
+private:
+    static constexpr int MAX_LOGS = 100;
+
+    std::thread mThread {};
+    moodycamel::ConcurrentQueue<LogData> mLogQueue { MAX_LOGS };
+    bool mAlive { true };
+
+    void Run();
+
+public:
+    RealTimeLogger();
+    ~RealTimeLogger();
+
+    template<typename... Args>
+    void Log(int level, const char *format, Args... args)
```

Comment:
you should consider making both of these (rtlogger::Log and normalLogger::Log) override some pure virtual interface. It will help them both match and changes will propagate to either of them

Reply:
I don't think I am able to make it a virtual function since it is a template https://stackoverflow.com/questions/2354210/can-a-class-member-function-template-be-virtual. Unless there is something I'm missing?

Reply:
wow! Today I learned, thanks for the info, I had no idea :) 

At dolby and spatial (the codebases where I spent the most time) they were very template-phobic. Templates are by far the weakest part of my C++ repertoire, love learning more about them!